### ISSUE TITLE
Doesn't work for local files 

### ISSUE THREAD

**Issue by Svish** [darkreader/darkreader#475]
When opening html and txt files directly from disk (or network share, if that matters), i.e. when the browser URL protocol is `file://`, then the dark mode does not work. No dark mode no matter how much I enable or disable or whitelist or blacklist. 😕

**Comment by alexanderby** [comment]
Try open `chrome://extensions`, find Dark Reader there, click **Allow access to file URLs**.

**Comment by Svish** [comment]
Would you look at that... that fixed it. So I guess actual title of this issue should be: Notify about that setting somehow in the extension?

**Comment by risualiser** [comment]
It's actually described [here](http://darkreader.org/help/).

**Comment by Svish** [comment]
Yeah, but there could be a notification the first time one opens a local file for example.

**Comment by Evpok** [comment]
Also this does not seem to work in firefox ?

**Comment by marcospb19** [comment]
Wait that's a option???????

Omg I've been searching for this for a long time.

What about this FAQ? When I click on the extension icon, there's no FAQ listed there, is the user supposed to search the website to find it?

IMHO this is not the best way to advertise this important feature.

**Comment by Gusted** [comment]
> What about this FAQ? When I click on the extension icon, there's no FAQ listed there, is the user supposed to search the website to find it?

Their is a "help" button in the UI. As we don't only provide a little FAQ but also some other explaination of other features as well, so the wording "help" was the best option.



> IMHO this is not the best way to advertise this important feature.

Uhm, the browser(only chrome) allows this after fuzzing around the settings of the extension. It's not that we blocks this feature intentionally.

**Comment by marcospb19** [comment]
The `Dev Tools` button is bigger than the `help` one.

Is it really as important as the GitHub link? The link to the FAQ?

I'm trying to be critic in a constructive way because I love this software, I've been using it for a LONG TIME, and I just wish the best for it, that said, I think that the UI should be more clear about this to the users that aren't being very attentive.

This "enable for local files" is generally more useful for the general user than `Dev Tools`.

**Comment by Gusted** [comment]
> The `Dev Tools` button is bigger than the `help` one.
> 
> Is it really as important as the GitHub link? The link to the FAQ?
> 
> I'm trying to be critic in a constructive way because I love this software, I've been using it for a LONG TIME, and I just wish the best for it, that said, I think that the UI should be more clear about this to the users that aren't being very attentive.
> 
> This "enable for local files" is generally more useful for the general user than `Dev Tools`.

Well, what's your opinion on the new UI?

Devtools -> Preview new Design.
![](https://i.imgur.com/K7aLV0n.png)
The help button is now under the settings button and bigger. All other related links are gone now(only the donate one for obvious reasons).

**Comment by marcospb19** [comment]
> Well, what's your opinion on the new UI?

GREAT!, I'm sorry for opening in an outdated interface, I was basing my thoughts in what I have here now.

**Comment by Gusted** [comment]
> GREAT!, I'm sorry for opining about an outdated interface, I was basing my thoughts in what I have here now.

No problems, it's still a preview and not the default one. But we are already tackling your issue with the new UI.

**Comment by sksmsWKd** [comment]
> Try open `chrome://extensions`, find Dark Reader there, click **Allow access to file URLs**.

Thank you save my eyes

**Comment by scarlion1** [comment]
> I've looked for a similar setting in Firefox and haven't found it. Is anyone aware if this is possible in a recent version of Firefox (eg: version > 80)?

In Firefox it seems that setting `security.fileuri.strict_origin_policy=false` (about:config) will allow Dark Reader to work on file:// URI's.  Dark Reader still says it's disabled for the current website but it's definitely enabled and applying the theme.  The toggle to enable/disable for the current website (file:// URI) doesn't work, and I can't change it from "Theme for all websites" to a custom theme for the current site, but the settings for the theme for all websites will be applied, at least.  Maybe firefox needs to be restarted for the add-on to fully work, but I'm in the middle of some other things and can't do a restart right now.

You can read about the implications of this setting [here](https://kb.mozillazine.org/Security.fileuri.strict_origin_policy).
